lib/clientv3/alphafeaturecheck.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientv3
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/projectcalico/libcalico-go/lib/apiconfig"
+	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+)
+
+// checkAlphaFeatures is used to check if GNP and NP rules are compatible with
+// alphaFeatures flag 
+func checkAlphaFeatures(alphaConfig string, ingress, egress []apiv3.Rule) error {
```

Comment:
I think this probably belongs in the validation package.

I know in the comment below we say that might be too much work for now, which I suppose is OK, but I'd still argue that even if we can't shoehorn it into the validator properly, we should at least expose this as its own public interface in the validation package and import it from there.

We can rename this `validateNoServiceAccountRules` and then the caller can decide whether or not to call this based on the enabled alpha features.

@robbrockbank @saumoh WDYT?